src/core/public/plugins/plugin_context.ts

```diff
@@ -40,24 +39,37 @@ export interface PluginInitializerContext {
     mode: Readonly<EnvironmentMode>;
     packageInfo: Readonly<PackageInfo>;
   };
+  readonly config: {
+    get: <T = ConfigSchema>() => T;
+  };
 }
 
 /**
  * Provides a plugin-specific context passed to the plugin's construtor. This is currently
  * empty but should provide static services in the future, such as config and logging.
  *
  * @param coreContext
- * @param pluginManinfest
+ * @param opaqueId
+ * @param pluginManifest
+ * @param pluginConfig
  * @internal
  */
 export function createPluginInitializerContext(
   coreContext: CoreContext,
   opaqueId: PluginOpaqueId,
-  pluginManifest: DiscoveredPlugin
+  pluginManifest: DiscoveredPlugin,
+  pluginConfig: {
+    [key: string]: unknown;
+  }
 ): PluginInitializerContext {
   return {
     opaqueId,
     env: coreContext.env,
+    config: {
+      get<T>() {
```

Comment:
`T extends object`? at least to outline the default value and requirements to the generic type